fix_includes.py

```diff
@@ -1944,7 +2016,9 @@ def _GetSymbolNameFromForwardDeclareLine(line):
   """
   iwyu_namespace_re = re.compile(r'namespace ([^{]*) { ')
   symbolname_re = re.compile(r'([A-Za-z0-9_]+)')
-  namespaces_in_line = iwyu_namespace_re.findall(line)
+  # Turn anonymous namespaces into their proper symbol representation.
+  namespaces_in_line = iwyu_namespace_re.findall(line.replace(
```

Comment:
For this line, you had mentioned that there were no tests stressing this but if you remove this line and run the tests that I've written, you should see two failures where forward declares are not being added to anonymous namespaces, due to the regex not taking those into account, leading to a bug where a symbol is not able to be found.  This does not happen at global scope because the previous function only worked when the namespace was the first one found so I don't know if I can break this into a seperate commit with it's own test without the rest of the changes.  Thoughts?

Reply:
I had to twiddle with this for a while until I saw what was going on. `_GetSymbolNameFromForwardDeclareLine` is only used for uniquing. This reminds me of Clang's USR concept --
```
/**
 * Retrieve a Unified Symbol Resolution (USR) for the entity referenced
 * by the given cursor.
 *
 * A Unified Symbol Resolution (USR) is a string that identifies a particular
 * entity (function, class, variable, etc.) within a program. USRs can be
 * compared across translation units to determine, e.g., when references in
 * one translation refer to an entity defined in another translation unit.
 */
```
(from https://github.com/llvm-mirror/clang/blob/master/include/clang-c/Index.h#L4153).

Since the test has `NsBaz` in three different namespaces, when nested anonymous namespaces were omitted, the symbol names would come out identical.

Let's not worry about the separate commit, but I wonder if we could make it clearer that this symbol name is only used for deduplication and not for display or anything. I can try and remember to address this after merge. Let's leave this as-is.